File: tests/fixtures/cpp/user_repository.cpp
#include "user_repository.hpp"

namespace app {

UserRepository::UserRepository() {}

std::string UserRepository::find(int id) const {
    auto it = users_.find(id);
    if (it != users_.end()) {
        return it->second;
    }
    return "";
}

void UserRepository::save(int id, const std::string& name) {
    users_[id] = name;
}

size_t UserRepository::count() const {
    return users_.size();
}

UserRepository build_default_repository() {
    UserRepository repo;
    repo.save(1, "alice");
    repo.save(2, "bob");
    return repo;
}

} // namespace app

File: tests/fixtures/rust/user_repository.rs
use std::collections::HashMap;

pub struct UserRepository {
    users: HashMap<u32, String>,
}

impl UserRepository {
    pub fn new() -> Self {
        UserRepository {
            users: HashMap::new(),
        }
    }

    pub fn find(&self, id: u32) -> Option<&String> {
        self.users.get(&id)
    }

    pub fn save(&mut self, id: u32, name: String) {
        self.users.insert(id, name);
    }

    pub fn count(&self) -> usize {
        self.users.len()
    }
}

pub fn build_default_repository() -> UserRepository {
    let mut repo = UserRepository::new();
    repo.save(1, "alice".to_string());
    repo.save(2, "bob".to_string());
    repo
}
